marlowe-playground-client/src/JavascriptEditor/Types.purs

```diff
@@ -0,0 +1,10 @@
+module JavascriptEditor.Types where
+
+import Language.Javascript.Interpreter as JS
+import Marlowe.Semantics (Contract)
+
+data JSCompilationState
+  = JSNotCompiled
+  | JSCompiling
+  | JSCompilationError JS.CompilationError
+  | JSCompiledSuccessfully (JS.InterpreterResult Contract)
```

Comment:
Let's just take a moment to laughcry at the notion of successfully compiled JavaScript.

Reply:
Also, I note this is an isomorphism for `RemoteData`.

Reply:
TBF it is actually TypeScript which we are pretending for marketing purposes is javascript

Reply:
Interesting marketing. 😂

Reply:
oh, do you think it's ok to use `RemoteData`, given the name?